src/ValueHandler/ImageValueHandler.php

```diff
@@ -89,7 +89,7 @@ public function handle($subject, string $attribute, array $value): void
             Assert::isInstanceOf($productImage, ProductImageInterface::class);
             /** @var ProductImageInterface $productImage */
             $productImage->setType($this->syliusImageType);
-            $productImage->addProductVariant($subject);
+            $subject->addImage($productImage);
```

Comment:
I changed this because apparently this does more in Sylius than only adding the variant to the image (and the variant should not only be added to the image but also the other way round).